tests/stress/stress_test.go

```diff
@@ -1,39 +1,298 @@
+		c.NumEnvironments = ENVIRONMENT_SCALE_MIN
+	}
 
-	tfContext := tftest.NewTestContext(t,
-		tftest.Dir(dir),
+	vars := map[string]string{
+		"endpoint":            config.APIEndpoint(),
+		"token":               config.AuthToken(),
+		"num_deploys":         strconv.Itoa(c.NumDeploys),
+		"num_deploy_families": strconv.Itoa(c.NumDeployFamilies),
+		"num_environments":    strconv.Itoa(c.NumEnvironments),
+		"num_load_balancers":  strconv.Itoa(c.NumLoadBalancers),
+		"num_services":        strconv.Itoa(c.NumServices),
+	}
+
+	terraform := tftest.NewTestContext(
+		b,
+		tftest.Dir("module"),
 		tftest.Vars(vars),
 		tftest.DryRun(*dry),
-		tftest.Log(log))
+		tftest.Log(b),
+	)
+
+	layer0 := clients.NewLayer0TestClient(b, vars["endpoint"], vars["token"])
+
+	terraform.Apply()
+	defer terraform.Destroy()
+
+	methodsToBenchmark := map[string]func(){}
 
-	layer0 := clients.NewLayer0TestClient(t, vars["endpoint"], vars["token"])
+	if c.NumDeploys > 0 {
+		deployIDs := strings.Split(terraform.Output("deploy_ids"), ",\n")
+		methodsToBenchmark["GetDeploy"] = func() { layer0.GetDeploy(deployIDs[0]) }
+		methodsToBenchmark["ListDeploys"] = func() { layer0.ListDeploys() }
+	}
 
-	// download modules using terraform get
-	tfContext.Terraformf("get")
+	if c.NumEnvironments > 0 {
+		environmentIDs := strings.Split(terraform.Output("environment_ids"), ",\n")
+		methodsToBenchmark["GetEnvironment"] = func() { layer0.GetEnvironment(environmentIDs[0]) }
+		methodsToBenchmark["ListEnvironments"] = func() { layer0.ListEnvironments() }
+	}
 
-	return &StressTest{
-		Terraform: tfContext,
-		Layer0:    layer0,
+	if c.NumLoadBalancers > 0 {
+		loadBalancerIDs := strings.Split(terraform.Output("load_balancer_ids"), ",\n")
+		methodsToBenchmark["GetLoadBalancer"] = func() { layer0.GetLoadBalancer(loadBalancerIDs[0]) }
+		methodsToBenchmark["ListLoadBalancers"] = func() { layer0.ListLoadBalancers() }
 	}
+
+	if c.NumServices > 0 {
+		serviceIDs := strings.Split(terraform.Output("service_ids"), ",\n")
+		methodsToBenchmark["GetService"] = func() { layer0.GetService(serviceIDs[0]) }
+		methodsToBenchmark["ListServices"] = func() { layer0.ListServices() }
+	}
+
+	if c.NumTasks > 0 {
+		methodsToBenchmark["ListTasks"] = func() { layer0.ListTasks() }
+
+		deployIDs := strings.Split(terraform.Output("deploy_ids"), ",\n")
+		environmentIDs := strings.Split(terraform.Output("environment_ids"), ",\n")
+
+		for i := 0; i < c.NumTasks; i++ {
+			environmentID := environmentIDs[i%len(environmentIDs)]
+			deployID := deployIDs[i%len(deployIDs)]
+			taskName := fmt.Sprintf("tt%v", i)
+			layer0.CreateTask(taskName, environmentID, deployID, 1, nil)
+		}
+	}
+
+	benchmark(b, methodsToBenchmark)
+}
+
+func benchmark(b *testing.B, methods map[string]func()) {
+	for name, fn := range methods {
+		b.Run(name, func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
+				fn()
+			}
+		})
+	}
+}
+
+func BenchmarkMinFamiliesMinDeploys(b *testing.B) {
```

Comment:
This may just be personal preference, but I don't think deploys are "interesting" enough to warrant this amount of test cases. Maybe we just have 3 tests?
```
MinFamiliesMinDeploys
MedFamiliesMedDeploys
MaxFamiliesMaxDeploys
```

Reply:
Sounds fine to me. 

✅